aws/ec2/monitorInstances.go

```diff
@@ -0,0 +1,284 @@
+	}
+)
+
+// merge function from https://blog.golang.org/pipelines#TOC_4
+// It allows to merge many chans to one.
+func merge(cs ...<-chan InstanceInfo) <-chan InstanceInfo {
+	var wg sync.WaitGroup
+	out := make(chan InstanceInfo)
+
+	// Start an output goroutine for each input channel in cs. The output
+	// copies values from c to out until c is closed, then calls wg.Done.
+	output := func(c <-chan InstanceInfo) {
+		for n := range c {
+			out <- n
+		}
+		wg.Done()
+	}
+	wg.Add(len(cs))
+	for _, c := range cs {
+		go output(c)
+	}
+
+	// Start a goroutine to close out once all the output goroutines are
+	// done. This must start after the wg.Add call.
+	go func() {
+		wg.Wait()
+		close(out)
+	}()
+	return out
+}
+
+// getAccountId gets the AWS Account ID for the given credentials
+func getAccountId(ctx context.Context, sess *session.Session) (string, error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	svc := sts.New(sess)
+	res, err := svc.GetCallerIdentity(nil)
+	if err != nil {
+		logger.Error("Error when getting caller identity", err.Error())
+		return "", err
+	}
+	return aws.StringValue(res.Account), nil
+}
+
+// getInstanceTag formats []*ec2.Tag to map[TagName]TagValue
+func getInstanceTag(tags []*ec2.Tag) map[TagName]TagValue {
+	res := make(map[TagName]TagValue)
+	for _, tag := range tags {
+		res[TagName(aws.StringValue(tag.Key))] = TagValue(aws.StringValue(tag.Value))
+	}
+	return res
+}
+
+func getCurrentCheckedHour() (start time.Time, end time.Time) {
+	now := time.Now()
+	end = time.Date(now.Year(), now.Month(), now.Day(), now.Hour()-1, 0, 0, 0, now.Location())
+	start = time.Date(now.Year(), now.Month(), now.Day(), now.Hour()-2, 0, 0, 0, now.Location())
+	return start, end
+}
+
+// getInstanceStats gets the CPU average and the CPU peak from CloudWatch
+func getInstanceStats(ctx context.Context, instanceId string,
+	sess *session.Session, region string) (float64, float64) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	start, end := getCurrentCheckedHour()
+	svc := cloudwatch.New(sess)
+	dimensions := []*cloudwatch.Dimension{
+		&cloudwatch.Dimension{
+			Name:  aws.String("InstanceId"),
+			Value: aws.String(instanceId),
+		},
+	}
+	stats, err := svc.GetMetricStatistics(&cloudwatch.GetMetricStatisticsInput{
+		Namespace:  aws.String("AWS/EC2"),
+		MetricName: aws.String("CPUUtilization"),
+		StartTime:  aws.Time(start),
+		EndTime:    aws.Time(end),
+		Period:     aws.Int64(int64(86400 * 30)),
+		Statistics: []*string{aws.String("Average"), aws.String("Maximum")},
+		Dimensions: dimensions,
+	})
+	if err != nil {
+		logger.Error("Error when fetching CPU stats from CloudWatch", err.Error())
+		return 0, 0
+	} else if len(stats.Datapoints) > 0 {
+		return aws.Float64Value(stats.Datapoints[0].Average), aws.Float64Value(stats.Datapoints[0].Maximum)
+	} else {
+		return 0, 0
+	}
+}
+
+// fetchInstancesList, firstly, calls DescribeInstances from AWS to list all the instances in a region.
```

Comment:
This type of comment does not really follow the coding style for Go comments. The first line should be a description of the function. Here the first line is not a complete description of what the function does